torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -699,32 +699,57 @@ Operation listEq(const Node* node) {
   };
 }
 
+template <typename T>
+Operation listNe(const Node* node) {
+  return [=](Stack& stack) {
+    T a;
+    T b;
+    pop(stack, a, b);
+    push(stack, !(a->elements() == b->elements()));
+    return 0;
+  };
+}
+
+inline bool tensor_list_equal(Shared<TensorList> a, Shared<TensorList> b) {
+  if (a->elements().size() != b->elements().size()) {
+    return false;
+  }
+
+  for (size_t i = 0; i < a->elements().size(); ++i) {
+    const auto& a_element = a->elements()[i];
+    const auto& b_element = b->elements()[i];
+    // This preserves Python's semantics, which uses eq() to compare two
+    // elements, then passes the result to bool().
+    // see: https://docs.python.org/3.4/reference/datamodel.html#object.__ge__
+    const auto cmp_result = a_element.eq(b_element);
+    if (!cmp_result.is_nonzero()) {
```

Comment:
That doesn't really conform to Python semantics, which will throw if the tensors weren't scalar tensors.